sycl/source/detail/scheduler/graph_processor.cpp

```diff
@@ -32,7 +32,6 @@ Scheduler::GraphProcessor::getWaitList(EventImplPtr Event) {
 
 void Scheduler::GraphProcessor::waitForEvent(EventImplPtr Event) {
   Command *Cmd = getCommand(Event);
-  assert(!Cmd && "Event has no associated command?");
```

Comment:
Could you clarify in the commit message the reason why this assertion is erroneous, please?
Test case would be great too.

Reply:
It seems there is just a typo in assert condition:
Should be: assert(Cmd && "Event has no associated command?");

Reply:
Right, fixed.

Reply:
I think it doesn't make sense to write test on this because the problems reproduces on any test which submits commands with asserts enabled.

Reply:
> I think it doesn't make sense to write test on this because the problems reproduces on any test which submits commands with asserts enabled.

Are you trying to say that this code is already covered by existing LIT tests?

Reply:
> Are you trying to say that this code is already covered by existing LIT tests?

Yes, this assertion leads to about 30 LIT test failures in debug mode without this patch.

Reply:
When asserts are enabled*